src/main/kotlin/org/wycliffeassociates/otter/common/domain/md/ParseMd.kt

```diff
@@ -0,0 +1,54 @@
+package org.wycliffeassociates.otter.common.domain.md
+
+import java.io.BufferedReader
+
+// TODO: Add Help type enum to HelpResource? (tn, tq)
+data class HelpResource(var title: String, var body: String)
+
+class ParseMd {
+
+    companion object {
+
+        private val isTitleRegex = Regex("^#+\\s*[^#\\s]+")
+        private val titleTextRegex = Regex("^#+\\s*")
+
+        fun parse(reader: BufferedReader): List<HelpResource> {
+
+            val helpResourceList = ArrayList<HelpResource>()
+
+            reader.forEachLine {
+
+                if (it.isEmpty())
+                    return@forEachLine // continue
+
+                // If we have a title, add a new help resource to the end of the list
+                if (isTitleLine(it)) {
+
+                    val titleText = getTitleText(it)
+                    helpResourceList.add(HelpResource(titleText, ""))
+                }
+                // Found body text. Add the body to the help resource at the end of the list
+                // If the list is empty, the body text will be discarded.
+                else if (helpResourceList.size > 0) {
+
+                    if (helpResourceList.last().body != "") {
+                        helpResourceList.last().body += " "
+                    }
+                    helpResourceList.last().body += it
+                }
+            }
+
+            return helpResourceList
+        }
+
+        internal fun getTitleText(line: String): String {
+
+            return line.removePrefix(titleTextRegex.find(line)!!.value)
```

Comment:
I know you check first to see if the title matches before you call this function, so you WILL match here; but it would still be better to refactor this to avoid the null, in the event that in the future this function is used and the if check is not done before calling it.